janis_core/tool/test_definitions.py

```diff
@@ -0,0 +1,197 @@
+        """
+        if tool.friendly_name() is None:
+            return "Missing friendly name"
+
+        return True
+
+    @staticmethod
+    def evaluate_metadata(tool: Tool) -> Union[str, bool]:
+        """
+        Evaluate if important metadata for documentation is p
+        """
+        METADATA_KEY_CONTRIBUTORS = "contributors"
+        METADATA_KEY_CREATED_DATE = "created date"
+        METADATA_KEY_INSTITUTION = "institution"
+
+        if isinstance(tool.metadata, ToolMetadata):
+            required = {
+                METADATA_KEY_CONTRIBUTORS: tool.metadata.contributors,
+                METADATA_KEY_CREATED_DATE: tool.metadata.dateCreated,
+                METADATA_KEY_INSTITUTION: tool.metadata.institution,
+            }
+
+            missing = []
+            for key, field in required.items():
+                if field is None or not field:
+                    missing.append(key)
+
+            # special case, tool_provider() value overwrites contributors in the documentation
+            if METADATA_KEY_INSTITUTION in missing:
+                if tool.tool_provider():
+                    missing.remove(METADATA_KEY_INSTITUTION)
+
+            if missing:
+                return f"Missing metadata: {', '.join(missing)}"
+        # elif isinstance(self.metadata, ...):
+        else:
+            return "Incorrect metadata class"
+
+        return True
+
+    @staticmethod
+    def evaluate_container(tool: Tool) -> Union[str, bool]:
+        """
+        Evaluate if the image specified for this tool exists in the remote registry
+        """
+        # If there is no container, we don't need to check if the container exists in the registry
+        if tool.container() is None:
+            return True
+
+        min_version_required = test_helpers.janis_assistant_version_required_min
+
+        try:
+            import janis_assistant
+
+            if parse_version(janis_assistant.__version__) < parse_version(
+                min_version_required
+            ):
+                raise Exception()
+
+            from janis_assistant.data.container.registries import (
+                ContainerRegistryBase,
+                ContainerRegistry,
+            )
+            from janis_assistant.data.container.info import ContainerInfo
+
+        except Exception as e:
+            raise Exception(
+                f"to run this test, janis_asisstant >= {min_version_required}"
+                f" must be installed"
+            )
+
+        # Call this outside the try-except so that we can still throw
+        # different exceptions relevant to the actual logic of this function
+        if not tool.container():
+            print(tool.id(), "no container")
+        else:
+            print(tool.id(), tool.container())
+
+        ci = ContainerInfo.parse(tool.container())
+        registry = ContainerRegistry.from_host(ci.host).to_registry()
+        digest = registry.get_digest(ci)
```

Comment:
https://github.com/PMCC-BioinformaticsCore/janis-assistant/blob/janis-prepare/janis_assistant/data/container/__init__.py#L29

Reply:
use the cache directory